app/src/org/commcare/fragments/personalId/PersonalIdPhoneVerificationFragment.java

```diff
@@ -126,34 +126,36 @@ public void onFailure(OtpErrorType errorType, @Nullable String errorMessage) {
 
             @Override
             public void onPersonalIdApiFailure(
-                    @NonNull BaseApiHandler.PersonalIdOrConnectApiErrorCodes failureCode, Throwable t) {
+                    @NonNull BaseApiHandler.PersonalIdOrConnectApiErrorCodes failureCode,
+                    Throwable t
+            ) {
                 if (handleCommonSignupFailures(failureCode)) {
                     return;
                 }
                 String error = PersonalIdApiErrorHandler.handle(activity, failureCode, t);
-                if(failureCode == BaseApiHandler.PersonalIdOrConnectApiErrorCodes.FAILED_AUTH_ERROR) {
+                if (failureCode == BaseApiHandler.PersonalIdOrConnectApiErrorCodes.FAILED_AUTH_ERROR) {
                     error = getString(R.string.personalid_incorrect_otp);
                 }
                 displayOtpError(error);
                 binding.connectPhoneVerifyButton.setEnabled(false);
             }
         };
-        otpManager = new OtpManager(activity, personalIdSessionData, otpCallback);
+        setupOtpManager(false);
     }
 
     /**
      * Determines if we should auto-switch from Firebase to PersonalId auth based on the error type.
      * Only switches from Firebase to PersonalId, never the reverse.
-     * 
+     *
      * @param errorType The OTP error type from Firebase
      * @return true if we should switch to PersonalId auth, false otherwise
      */
     private boolean shouldAutoSwitchToPersonalIdAuth(OtpErrorType errorType) {
-        if(SMS_METHOD_PERSONAL_ID.equalsIgnoreCase(personalIdSessionData.getSmsMethod())) {
+        if (SMS_METHOD_PERSONAL_ID.equalsIgnoreCase(personalIdSessionData.getSmsMethod())) {
             return false;
         }
```

Comment:
think this can also be part of `setupOtpManager` to decide on `allowedToFallback`. The idea here is if the default method is already `SMS_METHOD_PERSONAL_ID` and then fall back logic is redundant as we are already using the fallback as default. Think we can now get rid of `shouldAutoSwitchToPersonalIdAuth` method and directly use `errorType.isNonRecoverable()` in it's place.

Reply:
I think this could work - I want to spend some more time testing this before I push changes for it